src/js/node/Fs.hx

```diff
@@ -683,12 +709,13 @@ extern class Fs {
 	/**
 		Asynchronous rmdir(2).
 	**/
-	static function rmdir(path:FsPath, callback:Error->Void):Void;
+	@:overload(function(path:FsPath, callback:Error->Void):Void {})
+	static function rmdir(path:FsPath, options:FsRmdirOptions, callback:Error->Void):Void;
```

Comment:
Any particular reason you used an overload here instead of making the argument optional?

Reply:
Because I don't master Haxe well enough yet 😄 
It is possible to have an optional argument followed by a required argument?!?

Reply:
Yes, and not just for externs. :)

Reply:
I've tried with an overload and with an optional argument: the resulting code is not the same.

```js
// Haxe call: Fs.rmdir("path/to/dir", function(err) {})

// With: static function rmdir(path:FsPath, callback:Error->Void):Void;
fs.rmdir("path/to/dir", function(err) {})

// With: static function rmdir(path:FsPath, ?options:FsRmdirOptions, callback:Error->Void):Void;
fs.rmdir("path/to/dir", null, function(err) {})
```

With the optional argument, the value `null` is passed: that's not exactly the same thing as omitting the argument.
Under certain circonstances, that can make a difference in native JS code (i.e. if the code checks for an `undefined` value instead of an "empty" value).

So I think overloads are better when dealing with externs.

Reply:
Fair enough. Yeah, Haxe doesn't differentiate between `null` and `undefined`.